assembl/views/api2/votes.py

```diff
@@ -196,3 +200,56 @@ def vote_results_csv(request):
     ctx._instance.csv_results(output, histogram)
     output.seek(0)
     return Response(body_file=output, content_type='text/csv')
+
+
+@view_config(context=InstanceContext, request_method='GET',
+             ctx_instance_class=VotingWidget,
+             name="vote_results_csv", permission=P_DISC_STATS)
+def global_vote_results_csv(request):
+    ctx = request.context
+    user_id = request.authenticated_userid
+    if not user_id:
+        raise HTTPUnauthorized
+    widget = ctx._instance
+    if widget.activity_state != "ended":
+        permissions = get_permissions(user_id, ctx.get_discussion_id())
+        if P_ADMIN_DISC not in permissions:
+            raise HTTPUnauthorized()
+    user_prefs = LanguagePreferenceCollection.getCurrent()
+    ideas = widget.db.query(Idea
+        ).join(AbstractIdeaVote, AbstractIdeaVote.idea_id==Idea.id
+        ).join(AbstractVoteSpecification
+        ).filter(AbstractVoteSpecification.widget_id==widget.id
+        ).distinct().all()
+    idea_ids = [i.id for i in ideas]
+    titles = [(idea.title.best_lang(user_prefs).value, idea.id) for idea in ideas]
+    titles.sort()
+    q = widget.db.query(Idea.id).filter(Idea.id.in_(idea_ids))
+    specs = [(spec.title.best_lang(user_prefs).value if spec.title else str(spec.id), spec)
+             for spec in widget.vote_specifications]
+    specs.sort()
+    for (t, spec) in specs:
+        a = aliased(spec.get_vote_class(), name="votes_%d"%spec.id)
+        q = q.outerjoin(a, (a.idea_id==Idea.id) & (a.vote_spec_id==spec.id))
+        q = q.add_columns(func.sum(a.vote_value).label('vsum_%d' % spec.id),
+                          func.count(a.id).label('vcount_%d' % spec.id))
+    q = q.group_by(Idea.id)
+    r = q.all()
+    r = {x[0]: x for x in r}
+    output = StringIO()
+    csvw = csv.writer(output)
+    csvw.writerow([""]+[t.encode('utf-8') for (t, spec) in specs])
+    for title, idea_id in titles:
+        row = [title.encode('utf-8')]
+        sourcerow = r[idea_id][1:]
+        for i, (t, spec) in enumerate(specs):
```

Comment:
Can we get a small comment on the counting? Reading what `r` will produce and following the algo along without running it is non-trivial.